components/ws-manager-mk2/main.go

```diff
@@ -115,25 +111,32 @@ func main() {
 		setupLog.Error(nil, "namespace cannot be empty")
 		os.Exit(1)
 	}
+
 	if cfg.Manager.SecretsNamespace == "" {
 		setupLog.Error(nil, "secretsNamespace cannot be empty")
 		os.Exit(1)
 	}
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:                 scheme,
-		MetricsBindAddress:     cfg.Prometheus.Addr,
-		Port:                   9443,
-		HealthProbeBindAddress: cfg.Health.Addr,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "ws-manager-mk2-leader.gitpod.io",
-		NewCache:               cache.MultiNamespacedCacheBuilder([]string{cfg.Manager.Namespace, cfg.Manager.SecretsNamespace}),
+		Scheme:                        scheme,
+		MetricsBindAddress:            cfg.Prometheus.Addr,
+		Port:                          9443,
+		HealthProbeBindAddress:        cfg.Health.Addr,
+		LeaderElection:                true,
+		LeaderElectionID:              "ws-manager-mk2-leader.gitpod.io",
+		LeaderElectionReleaseOnCancel: true,
+		NewCache: func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
```

Comment:
Why the switch from the MultiNamespaceCache? This approach seems to do the same thing?

Reply:
I saw a deprecation warning with the old signature.